### PR TITLE
klee: 2.3 -> 3.0

### PR DESCRIPTION
**Pull request by McSinyx**
###### Description of changes

Upgrade to KLEE 3.  `ENABLE_KLEE_UCLIBC` was deprecated, and the CMake file now tries to be smart and deduce compiler paths from LLVM binaries so we need to hardcode path to clang instead.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by numinit — APPROVED** [review]
Thanks for this! Looks like klee-uclibc is still built and we just don't need to ask for it anymore.

File: pkgs/applications/science/logic/klee/default.nix [hunk 1]
```diff
@@ -45,30 +45,24 @@ let
...
     rev = "v${version}";
-    sha256 = "sha256-E1c6K6Q+LAWm342W8I00JI6+LMvqmULHZLkv9Kj5RmY=";
+    hash = "sha256-y5lWmtIcLAthQ0oHYQNd+ir75YaxHZR9Jgiz+ZUFQjY=";
   };
 
+  nativeBuildInputs = [ cmake ];
   buildInputs = [
     cryptominisat
     gperftools
-    lit # Configure phase checking for lit
```
Comment (numinit) [inline pkgs/applications/science/logic/klee/default.nix:60]: We should remove the `lit` function argument too (around line 16)

**Review by numinit — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/logic/klee/default.nix (hunk 1 above)
Comment (McSinyx) [reply pkgs/applications/science/logic/klee/default.nix:60]: It's still used in `nativeCheckInputs` below.

Comment (numinit) [reply pkgs/applications/science/logic/klee/default.nix:60]: Oops, somehow missed that. Sounds good.

**Review by numinit — APPROVED** [review]

**Pull request merged**